src/Common/CustomAttributes/CmdletPreviewAttribute.cs

```diff
@@ -25,19 +26,36 @@ public class CmdletPreviewAttribute : System.Attribute
     {
         public string _message;
 
+        public DateTime EstimatedGaDate { get; }
+
+        public bool IsEstimatedGaDateSet { get; } = false;
+
         public CmdletPreviewAttribute()
         {
             this._message = Resources.PreviewCmdletMessage;
         }
 
         public CmdletPreviewAttribute(string message)
         {
-            this._message = message;
+            this._message = string.IsNullOrEmpty(message) ? Resources.PreviewCmdletMessage : message;
+        }
+
+        public CmdletPreviewAttribute(string message, string estimatedDateOfGa) : this(message)
```

Comment:
How do we avoid confusion for dates like "1/2/2023"? Some may interpret it as Jan 2nd, some Feb 1st.

Reply:
The date assume the date follows "en-US" culture like breaking change attribute, which means dates are written in the month–day–year order.

Reply:
The benefits of using local culture to display estimated GA date is customer are more familiar with their local date. But considering:
- to keep consistent with breaking change attribute, CmdletPreviewAttribute won't display estimatedGaDate with local culture to avoid introduce new confusion.
- Customer may use the output as the input of DateTime, standard date time will avoid parse issue.